acme/acme/challenges.py

```diff
@@ -216,6 +215,74 @@ def response_and_validation(self, account_key, *args, **kwargs):
 
 
 @ChallengeResponse.register
+class DNS01Response(KeyAuthorizationChallengeResponse):
+    """ACME "dns-01" challenge response."""
+    typ = "dns-01"
+
+    def simple_verify(self, chall, domain, account_public_key):
+        """Simple verify.
+
+        :param challenges.DNS01 chall: Corresponding challenge.
+        :param unicode domain: Domain name being verified.
+        :param account_public_key: Public key for the key pair
+            being authorized.
+
+        :returns: ``True`` iff validation with the TXT records resolved from a
+            DNS server is successful.
+        :rtype: bool
+
+        """
+        if not self.verify(chall, account_public_key):
+            logger.debug("Verification of key authorization in response failed")
+            return False
+
+        validation_domain_name = chall.validation_domain_name(domain)
+        validation = chall.validation(account_public_key)
+        logger.debug("Verifying %s at %s...", chall.typ, validation_domain_name)
+
+        try:
+            from acme import dns_resolver
+        except ImportError:  # pragma: no cover
+            raise errors.Error("Local validation for 'dns-01' challenges "
```

Comment:
Not sure.   My idea was that if you do call self-verification you should have the extra library installed (i.e., code using the client should ensure the lib is installed).  Returning false means "I tried the verification and it failed".   That is not the case if the library is missing, in that case a precondition for the call is missing.

A "typical" client would have to abort trying to answer the challenge (assuming it will fail).  And the client would not be able to determine if it is a problem that is solvable by creating a new challenge and trying to verify it.

Thoughts?